extensions/file-manager/src/utils.tsx

```diff
@@ -1,140 +1,131 @@
-import {
-  ActionPanel,
-  CopyToClipboardAction,
-  List,
-  ShowInFinderAction,
-  OpenWithAction,
-  OpenAction,
-  showToast,
-  ToastStyle,
-  getPreferenceValues,
-  PushAction,
-  Icon,
-  Detail,
-} from "@raycast/api";
+import { ActionPanel, List, showToast, Toast, Icon, Action } from "@raycast/api";
 import { promisify } from "node:util";
 import { exec as _exec } from "node:child_process";
-import filesize from "filesize";
-import { existsSync, lstatSync, readdirSync, readlinkSync } from "node:fs";
+import { filesize } from "filesize";
+import { lstatSync, readdirSync, readlinkSync } from "node:fs";
 import { resolve } from "node:path";
 import { homedir } from "node:os";
+import { getExtPreferences } from "./preferences";
+import { ActionsGoDirectoryPairs } from "./utils/actions";
+import { Directory } from "./ui/Directory";
+import { FileDataType, FileType } from "./types";
 
 const exec = promisify(_exec);
 
-export type FileType = "directory" | "file" | "symlink" | "other";
-
-export type FileDataType = {
-  type: FileType;
-  name: string;
-  size: number;
-  permissions: string;
-  path: string;
-};
-
-export type PreferencesType = {
-  showDots: boolean;
-  directoriesFirst: boolean;
-  caseSensitive: boolean;
-  showFilePermissions: boolean;
-  showFileSize: boolean;
-  startDirectory: string;
-};
-
 export async function runShellScript(command: string) {
   const { stdout, stderr } = await exec(command);
   return { stdout, stderr };
 }
 
 export function getStartDirectory(): string {
-  let { startDirectory } = getPreferenceValues();
+  let { startDirectory } = getExtPreferences();
```

Comment:
`getExtPreferences` function is in this file `extensions/file-manager/src/preferences.ts`. 
The returned value of `getExtPreferences` function always has a correct TS type.